### DISCUSSION TITLE
How to Retain ParentId Information in grains-Grain2d Object

### DISCUSSION THREAD

**Discussion by lockensimon** [mtex-toolbox/mtex#2264] [Ask Anything]
Hello everyone, 

I am trying to store the information from ParentId within the grains-Grain2d object when merging my grains. The following code:

`[grains_merge, grains.prop.parentId] = merge(grains_smooth, DauphineBoundary);`

works partly (the parentId is stored under grains -> prop -> parentId), but it deletes all the information from grains.CS (at least the variable is empty afterward). So this seems not to be the right way.
When I use:

`[grains_merge, parentId] = merge(grains_smooth, DauphineBoundary);`

this doesn't happen and grains.CS is still intact. So the second code seems to be the way to go. Is it possible to store the parentId object in the grains object afterward?

Thank you very much!

**Comment by kilir** [comment]
Hi Simon,
not sure why this happens. For me storing the parentId doesn't seem to interfere.

```Matlab
mtexdata twins

[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd);

gB_MgMg = grains.boundary('m','m')

twinning = orientation.map(Miller(0,1,-1,-1,CS),Miller(-1,1,0,-1,CS),...
  Miller(1,0,-1,1,CS,'uvw'),Miller(1,0,-1,-1,CS,'uvw'))

twinBoundary = gB_MgMg(angle(gB_MgMg.misorientation,twinning) < 5*degree);

[grains_merge, grains.prop.parentId] = merge(grains, twinBoundary);

grains.CS

 ans = crystalSymmetry
 
   mineral        : Magnesium         
   color          : LightSkyBlue      
   symmetry       : 6/mmm             
   elements       : 24                
   a, b, c        : 3.2, 3.2, 5.2     
   reference frame: X||a*, Y||b, Z||c*

```
Cheers,

Rüdiger
